SmartDeviceLink/SDLCarWindow.m

```diff
@@ -0,0 +1,226 @@
+
+    return self;
+}
+
+- (void)syncFrame {
+    if (!self.streamManager.isVideoConnected || self.streamManager.isVideoStreamingPaused) {
+        return;
+    }
+
+    if (self.isLockScreenPresenting || self.isLockScreenDismissing) {
+        SDLLogD(@"Paused CarWindow, lock screen moving");
+        return;
+    }
+
+    CGRect bounds = self.rootViewController.view.bounds;
+
+    UIGraphicsBeginImageContextWithOptions(bounds.size, YES, 1.0f);
+    switch (self.renderingType) {
+        case SDLCarWindowRenderingTypeLayer: {
+            [self.rootViewController.view.layer renderInContext:UIGraphicsGetCurrentContext()];
+        } break;
+        case SDLCarWindowRenderingTypeView: {
+            [self.rootViewController.view drawViewHierarchyInRect:bounds afterScreenUpdates:self.drawsAfterScreenUpdates];
+        } break;
+    }
+
+    UIImage *screenshot = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+
+    CGImageRef imageRef = screenshot.CGImage;
+    CVPixelBufferRef pixelBuffer = [self.class sdl_pixelBufferForImageRef:imageRef usingPool:self.streamManager.pixelBufferPool];
+    [self.streamManager sendVideoData:pixelBuffer];
+    CVPixelBufferRelease(pixelBuffer);
+}
+
+#pragma mark - SDLNavigationLockScreenManager Notifications
+- (void)sdl_willPresentLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenPresenting = YES;
+}
+
+- (void)sdl_didPresentLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenPresenting = NO;
+}
+
+- (void)sdl_willDismissLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenBeingDismissed = YES;
+}
+
+- (void)sdl_didDismissLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenBeingDismissed = NO;
+}
+
+#pragma mark - SDLNavigationLifecycleManager Notifications
+- (void)sdl_didReceiveVideoStreamStarted:(NSNotification *)notification {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        // If the video stream has started, we want to resize the streamingViewController to the size from the RegisterAppInterface
+        self.rootViewController.view.frame = CGRectMake(0, 0, self.streamManager.screenSize.width, self.streamManager.screenSize.height);
+        self.rootViewController.view.bounds = self.rootViewController.view.frame;
+
+        SDLLogD(@"Video stream started, setting CarWindow frame to: %@", NSStringFromCGRect(self.rootViewController.view.bounds));
+    });
+}
+
+- (void)sdl_didReceiveVideoStreamStopped:(NSNotification *)notification {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        // And also reset the streamingViewController's frame, because we are about to show it.
+        self.rootViewController.view.frame = [UIScreen mainScreen].bounds;
+        SDLLogD(@"Video stream ended, setting view controller frame back: %@", NSStringFromCGRect(self.rootViewController.view.frame));
+    });
+}
```

Comment:
Another issue that I am having on reconnects is that sometimes when I try to get the topmost view controller from the `UIApplication`, it returns the `SDLScreenshotViewController`. I assume that is is because the `SDLScreenshotViewController` is not being dismissed when video stream is stopped.

Reply:
This turned out to be a user error on my part and not an issue with the library.

Reply:
What was the user error, is there anything we can document to help people if they encounter the same issue?

Reply:
It had to do with setting the `rootViewController` with an on-screen view controller at the wrong time. I added an in-depth description of the issue to a comment left on the `SDLCarWindow.m` class.

Reply:
Done